dss/error.py

```diff
@@ -37,6 +37,22 @@ def __init__(self, title: str = "User is not authorized to access this resource"
                          title,
                          *args, **kwargs)
 
+
+def maybe_fake_error(headers, code) -> bool:
```

Comment:
Why were probabilistic errors moved here? I think the previous mechanism with a single error (504) seems sufficient.

Reply:
First, to conform to the previous standard.  Since the mechanism is in place, it made sense to me to extend it.

Second, using `mock` doesn't raise the error here (where we want to test) but in `chalice/app.py`.

Third, using `mock` has difficulties.  Using one call throws the intended error and passes, but using a second call does not.

So I chose this because it's pretty straight forward, it uses an existing mechanism (which I need to use anyway to properly raise a `504`), and I think most importantly hits `dss/errors.py`, which is what should be propagating these errors.

Reply:
Let's please not extend the random errors, and not raise random errors in `dss_handler`.

It's OK to omit the probabilistic chalice exception from these tests for now. We can address it in the future.

Reply:
Alright then.